core/engine/src/module/mod.rs

```diff
@@ -181,6 +185,7 @@ impl Module {
     pub fn synthetic(
         export_names: &[JsString],
         evaluation_steps: SyntheticModuleInitializer,
+        path: Option<PathBuf>,
         realm: Option<Realm>,
         context: &mut Context,
     ) -> Self {
```

Comment:
Since this is a breaking change, I'd suggest creating a new `synthetic_with_path` method instead, which should be retrocompatible.

Reply:
Changed my mind. Let's keep it this way and make a breaking change on the next version to pass a `SyntheticModuleOptions` parameter instead. That would protect us from further breaking changes instead, and should be a bit more extensible.